CMakeLists.txt

```diff
@@ -23,10 +23,24 @@ include_directories(${catkin_INCLUDE_DIRS})
 link_directories(${catkin_LIBRARY_DIRS})
 
 # Qt Stuff
-find_package(Qt4 REQUIRED)
-include(${QT_USE_FILE})
+option(UseQt5 "UseQt5" OFF)
+if (UseQt5)
+  find_package(Qt5Widgets REQUIRED)
+  find_package(Qt5Core REQUIRED)
+  find_package(Qt5OpenGL REQUIRED)
+
+  # this can be removed once CMAKE 2.8.12 is available (jade)
+  # it is due to a change in Qts use of -fPIE and -fPIC
+  # http://code.qt.io/cgit/qt/qtbase.git/tree/dist/changes-5.4.2
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${Qt5Core_EXECUTABLE_COMPILE_FLAGS}")
+
+else (UseQt5)
+  find_package(Qt4 REQUIRED COMPONENTS QtCore QtGui QtOpenGL)
+  include(${QT_USE_FILE})
+endif(UseQt5)
```

Comment:
Just found the [CMake coding standards](http://docs.ros.org/jade/api/catkin/html/user_guide/standards.html) and will open another PR soonish.

Do you have a reference for the cmake part? The cmake documentation still uses this form: https://cmake.org/cmake/help/v3.5/command/if.html
>  Note that the expression in the else and endif clause is optional.

So it's optional but not deprecated nor is it recommended to omit it.